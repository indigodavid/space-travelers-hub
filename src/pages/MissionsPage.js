import React from 'react';
import Missions from '../components/Missions';
import '../styles/MissionsPage.scss';

const MissionsPage = () => (
  <div className="missions">
    <header>
      <h2>Missions</h2>
      <h2>Descriptions</h2>
      <h2>Status</h2>
    </header>
    <Missions />
  </div>
);

export default MissionsPage;
