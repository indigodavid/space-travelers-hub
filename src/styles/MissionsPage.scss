/* stylelint-disable at-rule-no-unknown */
@import 'App.scss';

@mixin missions-grid {
  display: grid;
  grid-template-columns: 1fr 3fr 2fr;
  gap: 1em;
}

@mixin flex-column-gap-1 {
  display: flex;
  flex-direction: column;
  gap: 1em;
}

.missions {
  @include flex-column-gap-1();

  margin: 1em auto;
  padding: 4%;
  background-color: $color-background;

  header {
    @include missions-grid();

    text-align: center;
  }
}

h2 {
  font-size: 1.5rem;
  font-weight: bold;
  margin: 0;
}

.missionsData {
  @include flex-column-gap-1();

  margin: 0 auto;

  li {
    @include missions-grid();

    border: $color-border solid 0.1em;
    border-radius: 0.5em;
    background-color: $color-negative;
    padding: 1em;

    .missionName {
      display: flex;
      justify-content: center;
      align-items: center;
      font-weight: 700;
      font-size: 1.1rem;
    }

    .description{
      display: flex;
      justify-content: center;
      align-items: center;
      font-size: 1rem;
      height: 150px;
      overflow: auto;
    }

    .status {
      display: flex;
      justify-content: center;
      align-items: center;
      gap: 1em;
    }

    button {
      appearance: none;
      border: none;
      background-color: $color-accent;
      cursor: pointer;
      color: $color-negative;
      font-size: 0.8rem;
      padding: 1em;
      border-radius: 0.5em;
      font-weight: bold;
      text-transform: uppercase;
      transition-duration: 0.4s;
      transition-timing-function: ease-in-out;

      &:hover {
        border: $color-accent solid 0.06em;
        background-color: $color-negative;
        color: $color-accent;
      }

      &.leave {
        background-color: $color-warning;
      }

      &.leave:hover {
        border-color: $color-warning;
        background-color: $color-negative;
        color: $color-warning;
      }
    }
  }
}

.membership {
  background-color: $color-disabled;
  color: $color-negative;
  font-size: 0.8rem;
  padding: 0.5em;
  text-transform: uppercase;
  border-radius: 0.5em;
  transition-duration: 0.1s;
  transition-timing-function: ease-in;

  &.active {
    background-color: $color-active;
    font-weight: bold;
  }
}
