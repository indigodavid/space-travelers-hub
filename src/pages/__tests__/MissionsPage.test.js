import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MissionsPage from '../MissionsPage';

const renderMissionsPage = () => {
  const { container } = render(
    <Provider store={store}>
      <MissionsPage />
    </Provider>,
  );
  return container;
};

describe('Test MissionsPage page', () => {
  test('Match Snapshot', () => {
    expect(renderMissionsPage().firstChild).toMatchSnapshot();
  });

  test('Test HTML elements', () => {
    expect(renderMissionsPage().querySelector('.missions')).toBeInTheDocument();
  });

  test('Find Elements ny text', () => {
    renderMissionsPage();
    expect(screen.getByText(/Status/i)).toBeInTheDocument();
    expect(screen.getByText(/Descriptions/i)).toBeInTheDocument();
  });

  test('Find buttons', () => {
    renderMissionsPage();
    // Wait for one second until data is retrieved from API
    setTimeout(() => {
      expect(screen.getByText(/Join Mission/i)).toBeInTheDocument();
    }, 1000);
  });
});
