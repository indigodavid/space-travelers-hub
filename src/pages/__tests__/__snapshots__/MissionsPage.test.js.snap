// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MissionsPage page Match Snapshot 1`] = `
<div
  class="missions"
>
  <header>
    <h2>
      Missions
    </h2>
    <h2>
      Descriptions
    </h2>
    <h2>
      Status
    </h2>
  </header>
  <ul
    class="missionsData"
  />
</div>
`;
